tsdb/head.go

```diff
@@ -519,6 +521,15 @@ func (h *Head) Init(minValidTime int64) error {
 		if _, ok := errors.Cause(err).(*chunks.CorruptionErr); ok {
 			h.metrics.mmapChunkCorruptionTotal.Inc()
 		}
+
+		if h.opts.EnableMemorySnapshotOnShutdown && !snapshotWasDiscarded {
```

Comment:
Will calling `h.resetInMemoryState()` twice cause any issues? If not the flag for if it was already discarded might not be necessary and we just always reset if an error is encountered.

Reply:
I think it is not gonna matter. And no worries about double allocation since this is a rare case. I will remove the boolean check.